server/src/com/thoughtworks/go/server/scheduling/BuildCauseProducerService.java

```diff
@@ -246,6 +251,46 @@ ServerHealthState newProduceBuildCause(PipelineConfig pipelineConfig, BuildType
         }
     }
 
+    private boolean isGoodReasonToSchedule(PipelineConfig pipelineConfig, BuildCause buildCause, BuildType buildType,
+                                           boolean materialConfigurationChanged,MaterialRevisions peggedRevisions) {
+        if(buildCause == null)
+            return false;
+
+        boolean validCause = buildType.isValidBuildCause(pipelineConfig, buildCause);
+
```

Comment:
Also, please get rid of the commented-out code and peggedRevisions parameter from the method, if its not required.